app/static/js/app.js

```diff
@@ -247,15 +250,50 @@ function onKeyDown(evt) {
 }
 
 function sendMouseEvent(evt) {
-  const boundingRect = evt.target.getBoundingClientRect();
-  const cursorX = Math.max(0, evt.clientX - boundingRect.left);
-  const cursorY = Math.max(0, evt.clientY - boundingRect.top);
+  // Ensure that e.g. mouse drags don't attempt to drag the image on the screen.
+  evt.preventDefault();
+
+  emitMouseEvent(evt.target, { x: evt.clientX, y: evt.clientY }, evt.buttons);
+}
+
+function sendTouchEvent(evt) {
+  // Ensure that drags don't attempt to drag the image on the screen.
+  evt.preventDefault();
+
+  const touches = evt.touches;
+
+  if (touches.length == 0 && lastPosition !== undefined) {
+    // Release all mouse buttons
+    socket.emit("mouse-event", {
```

Comment:
Would it be possible for us to refactor the code so that only one function calls `socket.emit("mouse-event"`? I'd like to generate all the `mouse-event` messages from a single location.